armi/bookkeeping/db/database3.py

```diff
@@ -1318,24 +1341,27 @@ def _packLocation(loc: grids.LocationBase) -> Tuple[str, List]:
     return locationType, locData
 
 
-def _unpackLocation(locationTypes, locData):
+def _unpackLocation(
+    locationTypes, locData, version: Tuple[int, int] = (DB_MAJOR, DB_MINOR)
+):
     """
     Convert location data as read from DB back into data structure for building reactor model.
 
     location and locationType will only have different lengths
     when multiindex locations are used.
     """
+    oldStyle = version[0] == 3 and version[1] < 3
     locsIter = iter(locData)
     unpackedLocs = []
     for lt in locationTypes:
-        if lt == LOC_NONE:
+        if (oldStyle and lt == "None") or lt == LOC_NONE:
             loc = next(locsIter)
             unpackedLocs.append(None)
-        elif lt == LOC_INDEX:
+        elif (oldStyle and lt == "IndexLocation") or lt == LOC_INDEX:
             loc = next(locsIter)
             # the data is stored as float, so cast back to int
             unpackedLocs.append(tuple(int(i) for i in loc))
-        elif lt == LOC_COORD:
+        elif oldStyle or lt == LOC_COORD:
             loc = next(locsIter)
             unpackedLocs.append(tuple(loc))
         elif lt.startswith(LOC_MULTI):
```

Comment:
Same as above; probably worth it to make two functions, or branch into {implementation for old}-{implementation for new}